graphwebhook/src/main/java/com/example/graphwebhook/GraphClientHelper.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.example.graphwebhook;
+
+import com.microsoft.graph.logger.DefaultLogger;
+import com.microsoft.graph.logger.LoggerLevel;
+import com.microsoft.graph.requests.GraphServiceClient;
+
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+
+import okhttp3.Request;
+
+public class GraphClientHelper {
+
+    /**
+     * private constructor to hide the implicit public one
+     */
+    private GraphClientHelper() {
+        throw new IllegalStateException("Static class");
+    }
+
+
+    /**
+     * @param oauthClient the authorized OAuth2 client to authenticate Graph requests with
+     * @return A Graph client object that uses the provided OAuth2 client for access tokens
+     */
+    public static GraphServiceClient<Request> getGraphClient(OAuth2AuthorizedClient oauthClient) {
+        final var authProvider = new SpringOAuth2AuthProvider(oauthClient);
```

Comment:
I'm curious to why we aren't using azure identity or at least msal here? is it to get integration with the dependency injection?

Reply:
oh interesting, I didn't know they'd recommend that, probably because they don't have an MS ID web equivalent yet.